File: Leetcode/deleteColumnsToMakeSorted.cpp
#include <bits/stdc++.h>

using namespace std;

int minDeletionSize(vector<string> &strs)
{
  int n = strs.size();
  int m = strs[0].size();
  int count = 0;
  for (int i = 0; i < m; i++)
  {
    for (int j = 0; j < n - 1; j++)
    {
      if (strs[j + 1][i] < strs[j][i])
      {
        count++;
        break;
      }
    }
  }
  return count;
}

int main()
{
  int n;
  cin >> n;
  vector<string> strs(n);
  for (int i = 0; i < n; i++)
    cin >> strs[i];

  cout << minDeletionSize(strs) << "\n";

  return 0;
}

File: Leetcode/Python/delete_columns_to_make_sorted.py
def main():
  strs = list(map(str, input().split()))
  print(minDeletionSize(strs))

def minDeletionSize(strs: list[str]) -> int:
  ans = 0
  n = len(strs)
  m = len(strs[0])

  for j in range(m):
    for i in range(n - 1):
      if strs[i][j] > strs[i + 1][j]:
        ans += 1
        break
  return ans

if __name__ == '__main__':
  main()